sigle/src/modules/settings/SettingsForm.tsx

```diff
@@ -128,15 +128,30 @@ export const SettingsForm = ({ settings, username }: SettingsFormProps) => {
       const website = formik.values.siteUrl;
 
       formik.values.siteTwitterHandle = handle.split('/').pop() as string;
-      newSettings.siteTwitterHandle = handle.split('/').pop() as string;
 
-      if (handle.includes('@')) {
-        formik.values.siteTwitterHandle = handle.replace('@', '');
-        newSettings.siteTwitterHandle = handle.replace('@', '');
-      }
+      formik.values.siteTwitterHandle = handle.replace('@', '');
+      newSettings.siteTwitterHandle = handle.replace('@', '');
+
+      const isValidUrl = (url: string) => {
+        try {
+          new URL(url);
+        } catch (e) {
+          console.error(e);
+          return false;
+        }
+        return true;
+      };
+
+      formik.values.siteUrl = isValidUrl(website)
+        ? website.replace(urlRegex, '')
+        : '';
+      newSettings.siteUrl = isValidUrl(website)
```

Comment:
prevents invalid input appearing on blog, despite formik error - using `formik.values.siteUrl` alone, it still shows whatever the user inputs. 

just wondering if it should be working without having to also set the `newSettings` value, or if I missed something with the logic?

Reply:
This logic could be moved in the `validate` section I think